Mathlib/LinearAlgebra/CliffordAlgebra/SpinGroup.lean

```diff
@@ -73,127 +73,110 @@ def inv_of_inv_ι (m : M) [Invertible (ι Q m)] [Invertible (2 : R)] : Invertibl
 
 #align invertible_of_invertible_ι inv_of_inv_ι
 
--- TODO: below are not fixed yet
+-- restore the lean3 behavior
+macro_rules | `($x * $y) => `(@HMul.hMul _ _ _ instHMul $x $y)
 
 /-- `lipschitz` is the subgroup closure of all the elements in the form of `ι Q m` where `ι`
 is the canonical linear map `M →ₗ[R] CliffordAlgebra Q`. -/
 def lipschitz (Q : QuadraticForm R M) :=
-  Subgroup.closure (coe ⁻¹' Set.range (ι Q) : Set (CliffordAlgebra Q)ˣ)
+  Subgroup.closure (Units.val ⁻¹' Set.range (ι Q) : Set (CliffordAlgebra Q)ˣ)
 #align lipschitz lipschitz
 
+set_option synthInstance.maxHeartbeats 25000 in
 /-- If x is in `lipschitz Q`, then `(ι Q).range` is closed under twisted conjugation. The reverse
 statement presumably being true only in finite dimensions.-/
 theorem mem_lipschitz_conjAct_le {x : (CliffordAlgebra Q)ˣ} [Invertible (2 : R)]
-    (hx : x ∈ lipschitz Q) : ConjAct.toConjAct x • (ι Q).range ≤ (ι Q).range :=
-  by
-  refine' @Subgroup.closure_induction'' _ _ _ _ _ hx _ _ _ _
-  · rintro x ⟨z, hz⟩ y ⟨a, ha⟩
-    simp only [SMul.smul, SetLike.mem_coe, LinearMap.mem_range, DistribMulAction.toLinearMap_apply,
-      ConjAct.ofConjAct_toConjAct] at ha
-    rcases ha with ⟨⟨b, hb⟩, ha1⟩
+    (hx : x ∈ lipschitz Q) : ConjAct.toConjAct x • LinearMap.range (ι Q) ≤ LinearMap.range (ι Q) := by
```

Comment:
@_**Eric Wieser|310045** [said](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Field.20notation.20and.20.60CoeFun.60/near/313393152):
> Having dot notation work on bundled morphisms like this was often convenient in Lean 3; it would be a shame to lose it
@_**Yaël Dillies|387244** [said](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/dot.20notation.20with.20Multiplicative/near/400138163):
> This is the general issue with bundled morphisms not taking dot notation anymore. This is really annoying.